server/bleep/src/webserver/answer.rs

```diff
@@ -534,18 +541,31 @@ impl Conversation {
 
             let explanations = proc_result
                 .lines
-                .iter()
+                .into_iter()
                 .filter(|r| r.start > 0 && r.end > 0)
-                .filter_map(|r| {
-                    let end = r.end.min(r.start + 10);
-
-                    Some(serde_json::json!({
-                        "start": r.start,
-                        "end": end,
-                        "relevant_code": lines
-                            .get(r.start.saturating_sub(1)..end.saturating_sub(1))?
+                .map(|mut r| {
+                    r.end = r.end.min(r.start + 20);
+                    r
+                })
+                .fold(Vec::<Range>::new(), |mut exps, next| {
+                    if let Some(prev) = exps.last_mut() {
+                        if prev.end >= next.start {
```

Comment:
We should merge ranges if the next range is within 10 lines of the next range. Something like:

```suggestion
                        if prev.end + 10 >= next.start {
```